### ISSUE TITLE
AWS cloud inspector detected vulnerability

### ISSUE THREAD

**Issue by kamaltherocky** [hashicorp/consul-template#633]
Hi,

I was running a vulnerability scan for one of the image in AWS and during the scan there was one reported against consul-template. The detail is as below

**Rules package** 
Runtime Behavior Analysis-1.0

**Finding** 
The following executable files on instance do not support stack cookies: /usr/local/bin/consul-template.

**Severity** Medium 

**Description** 
This rule detects the presence of third-party software that is compiled without support for stack cookies. Stack cookies increase system security by defending against stack-based buffer overflow and other memory corruption attacks.

**Recommendation** 
It is recommended that you uninstall this software from your assessment target if you are not using it, or contact the vendor to get an updated version of this software with stack cookies enabled.

**Comment by sethvargo** [comment]
Hi @kamaltherocky 

Thank you for opening an issue. In the future, _please_ follow the responsible disclosure policy: https://www.hashicorp.com/security.html.

I ran this test myself on a stock Consul Template installation on an Ubuntu 14.04 AMI and was not able to reproduce this assessment warning.

At first, I thought this might be due to the fact that CGO is enabled. However, after compiling a version of Consul Template with CGO enabled and a version with GCO enabled, the assessment warning was not raised on either instance. As such, that does not seem to be the root cause.

Another potential cause, which I was also not able to reproduce, is if the arbitrary "command" argument passed to the Consul Template processes executes a command which has this vulnerability. It's possible that Amazon's static analysis tool senses this leak is coming from the Consul Template process when it's actually the subprocess spawned when a template changes.

We've discussed this with our team, and there is likely nothing we can do to mitigate this in Consul Template. If this continues to happen, it may be a bug in Amazon's static analysis tool.
